freedb/protocols.md

```diff
@@ -1,8 +1,311 @@
+    value = models.StringField()
+```
+### Operations
+
+#### Set
+
+Set the value of a `key` to the given `value`.
+
+The operation method should expect these parameters:
+
+| Parameter Name | Required | Remarks                       |
+| -------------- | -------- | ----------------------------- |
+| `key`          | Yes      | Record's key                  |
+| `value`        | Yes      | The value that we want to set |
+
+
+For both Default and Append Only mode, simply insert `Entry(key=<key>, value=<value>)` to the row store.
+
+#### Get
+
+Get the value of a `key` from the store.
+
+The operation method should expect these parameters:
+
+| Parameter Name | Required | Remarks                          |
+| -------------- | -------- | -------------------------------- |
+| `key`          | Yes      | Record's key that we want to get |
+
+1. For Default Mode, `Get` can be done by selecting rows that matched `key=<key>` condition and applying `limit=1`.
+	- In `PyFreeDB` syntax, it's equivalent to: `row_store.select().where("key = ?", <key>).limit(1).execute()`.
+
+2. For Append-Only mode, `Get` can be done by selecting rows that matched `key=<key>` sorted by `_rid` value in
+   descending manner and applying `limit=1`. Note that if we get `value=""` we must treat it as `KeyNotFound`.
+	- In `PyFreeDB` syntax, it's equivalent to: `row_store.select().where("key = ?", key).order_by(Ordering.DESC("_rid")).limit(1).execute()`.
+
+
+#### Delete
+
+Deletes the record associated with the given `key` from the store.
+
+The operation method should expect these parameters:
+| Parameter Name | Required | Remarks                             |
+| -------------- | -------- | ----------------------------------- |
+| `key`          | Yes      | Record's key that we want to delete |
+
+1. For default mode, `Delete` can be done by deleting row that matched with `key=<key>`.
+	- In `PyFreeDB` syntax, it's equivalent to `row_store.delete().where("key = ?", key).execute()`.
+2. For Append-Only mode, `Delete` can be done by inserting `Entry(key=<key>, value="")` to the store.
+	- In `PyFreeDB` syntax, it's equivalent to `row_store.insert(Entry(key=key, value="")).execute()`
+
+
+[AppendAPI]: https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/append
+[BatchUpdateAPI]: https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/batchUpdate
+[ClearAPI]: https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/clear
 
-> As of 31 August 2022, this document still acts as a placeholder so it can be referred to from other documents.
-> We are still prioritising other things.
-> We will revisit and write the details as soon as possible.
\ No newline at end of file
+[GVizAPI]: https://developers.google.com/chart/interactive/docs/reference
+[A1Notation]: https://developers.google.com/sheets/api/guides/concepts
+[QueryLanguage]: https://developers.google.com/chart/interactive/docs/querylanguage
```

Comment:
What's the purpose of listing these links down? I think there should be a "references" section instead?

Reply:
To make the document source tidier i'm using [reference-style-links](https://www.markdownguide.org/basic-syntax/#reference-style-links),

Reply:
Oh the formatting is fine, but let's have a `References` section? I believe it makes it clearer that these links are for references (otherwise people think it's related to the `Delete` part above.

Reply:
> otherwise people think it's related to the Delete part above.

Oh I mean these links will not be shown in the rendered HTML.